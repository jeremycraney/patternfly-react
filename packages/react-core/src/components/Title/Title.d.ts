import { SFC, HTMLProps, ReactNode } from 'react';

export enum TitleSize {
  xxxxl = 'xxxxl',
  xxxl = 'xxxl',
  xxl = 'xxl',
  xl = 'xl',
  lg = 'lg',
  md = 'md'
}

export interface TitleProps extends HTMLProps<HTMLHeadingElement> {
  size: TitleSize;
  withMargins?: boolean;
  children?: ReactNode;
}

declare const Title: SFC<Props>;

export default Title;
