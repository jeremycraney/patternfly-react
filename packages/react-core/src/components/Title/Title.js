import React from 'react';
import { css, getModifier } from '@patternfly/react-styles';
import PropTypes from 'prop-types';
import styles from '@patternfly/patternfly-next/components/Title/styles.css';

export const TitleSize = {
  xxxxl: 'xxxxl',
  xxxl: 'xxxl',
  xxl: 'xxl',
  xl: 'xl',
  lg: 'lg',
  md: 'md'
};

const propTypes = {
  size: PropTypes.oneOf(Object.keys(TitleSize)).isRequired,
  children: PropTypes.node,
  className: PropTypes.string,
  withMargins: PropTypes.bool
};

const defaultProps = {
  children: '',
  className: '',
  withMargins: false
};

const Title = ({ size, withMargins, className, children, ...props }) => (
  <h1
    {...props}
    className={css(
      styles.title,
      getModifier(styles, size),
      withMargins && styles.modifiers.margin,
      className
    )}
  >
    {children}
  </h1>
);

Title.propTypes = propTypes;
Title.defaultProps = defaultProps;

export default Title;
