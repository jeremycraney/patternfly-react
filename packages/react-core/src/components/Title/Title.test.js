import Title, { TitleSize } from './Title';
import React from 'react';
import { shallow } from 'enzyme';

Object.values(TitleSize).forEach(size => {
  test(`${size} Title`, () => {
    const view = shallow(<Title size={size}>{size} Title</Title>);
    expect(view).toMatchSnapshot();
  });
});

test('hasMargin', () => {
  const view = shallow(
    <Title size="xl" withMargins>
      With margin
    </Title>
  );
  expect(view).toMatchSnapshot();
});
