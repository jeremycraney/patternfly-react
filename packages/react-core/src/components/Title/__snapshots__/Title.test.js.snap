// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hasMargin 1`] = `
.pf-c-title.pf-m-xl.pf-m-margin {
  display: block;
  font-size: 1.3125rem;
  font-weight: 400;
  line-height: 1.5;
  margin-top: 1.5rem;
  margin-bottom: 1rem;
}

<h1
  className="pf-c-title pf-m-xl pf-m-margin"
>
  With margin
</h1>
`;

exports[`lg Title 1`] = `
.pf-c-title.pf-m-lg {
  display: block;
  font-size: 1.125rem;
  font-weight: 600;
  line-height: 1.5;
}

<h1
  className="pf-c-title pf-m-lg"
>
  lg
   Title
</h1>
`;

exports[`md Title 1`] = `
.pf-c-title.pf-m-md {
  display: block;
  font-size: 1rem;
  font-weight: 600;
  line-height: 1.5;
}

<h1
  className="pf-c-title pf-m-md"
>
  md
   Title
</h1>
`;

exports[`xl Title 1`] = `
.pf-c-title.pf-m-xl {
  display: block;
  font-size: 1.3125rem;
  font-weight: 400;
  line-height: 1.5;
}

<h1
  className="pf-c-title pf-m-xl"
>
  xl
   Title
</h1>
`;

exports[`xxl Title 1`] = `
.pf-c-title.pf-m-xxl {
  display: block;
  font-size: 1.5rem;
  font-weight: 400;
  line-height: 1.3;
}

<h1
  className="pf-c-title pf-m-xxl"
>
  xxl
   Title
</h1>
`;

exports[`xxxl Title 1`] = `
.pf-c-title.pf-m-xxxl {
  display: block;
  font-size: 1.75rem;
  font-weight: 400;
  line-height: 1.3;
}

<h1
  className="pf-c-title pf-m-xxxl"
>
  xxxl
   Title
</h1>
`;

exports[`xxxxl Title 1`] = `
.pf-c-title.pf-m-xxxxl {
  display: block;
  font-size: 2.25rem;
  font-weight: 400;
  line-height: 1.3;
}

<h1
  className="pf-c-title pf-m-xxxxl"
>
  xxxxl
   Title
</h1>
`;
